### ISSUE TITLE
indent: [2, "tab"] - does not pick up single space

### ISSUE THREAD

**Issue by julesmoretti** [eslint/eslint#7000]
**What version are you using?**
"gulp-eslint": "^3.0.1",

**What did you do?**
**What happened?**
I tried to have only tabbed indentation but at the start of a function when I add 1 space it does not error out like so:

```
 const hello = () =>{
	console.log('thats it!' );
};

hello();
```

Recap:
```
(space)const hello...
(tab)console.log(...
};
```

The first space does not get returned as an error. This is my config file:

```
{
	"env": {
		"es6": true
	},
	"rules": {
		"indent": [2, "tab"]
	}
}

```

**What did you expect to happen?**

I expected Eslit to return an error due to that extra space before the function.

**Comment by platinumazure** [comment]
There's a known issue around the first line of a file defining the base indent for the rest.

Does your issue also reproduce if the space line is not the first line of code in the file?

```
(tab)someCode();
(space)indentLineErrorHere();  // does the indent error show up here?
(tab)noErrorHere();
```

**Comment by julesmoretti** [comment]
@platinumazure sorry for the delay. Yes it does:

See attachment:

![screen shot 2016-09-06 at 12 58 41 pm](https://cloud.githubusercontent.com/assets/744300/18288789/b503b8e2-7431-11e6-8bd1-7a19865ce2ef.png)

**Comment by platinumazure** [comment]
@julesmoretti Then it seems this is an issue only on the first line (where we presume that the first line might define the base indent for the rest of the file).

I'm not sure if we can change this as a bugfix-- we might need to make it an enhancement (i.e., add a new option that says assume first line indent should absolutely be zero).

Does that sound good to you?

**Comment by julesmoretti** [comment]
That sounds good to me @platinumazure thank you!
